src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs

```diff
@@ -18,7 +18,7 @@ private static Delegate CreateBuiltInDelegate (Delegate dlg, Type delegateType)
 		{
 			switch (delegateType.Name) {
 				case nameof (_JniMarshal_PP_V): {
-					_JniMarshal_PP_V callback = (_JniMarshal_PP_V) Delegate.CreateDelegate (typeof (_JniMarshal_PP_V), dlg.Target, dlg.Method);
+					_JniMarshal_PP_V callback = (_JniMarshal_PP_V) dlg;
```

Comment:
Some tests hit:
```
System.InvalidCastException : Specified cast is not valid.
```
Thinking about this... I think `_JniMarshal_PP_V` is actually private to each assembly, and so this might not actually work?

If you have a class library that binds a Java library for C#, it gets its own `internal` version of these delegates. So if this code is in `Mono.Android.dll`, it wouldn't be able to cast as-is.

Reply:
Will check if this works:
```csharp
var callback = Unsafe.As<_JniMarshal_PP_V> (dlg);
```
The signature should match, seems like it might work?

Reply:
Another thought would be to do an `is _JniMarshal_PP_V` check, and if it isn't, then call `Delegate.CreateDelegate(typeof(_JniMarshal_PP_V))`.